pkgs/development/tools/build-managers/apache-maven/build-package.nix

```diff
@@ -26,13 +29,28 @@ let
       maven
     ];
 
-    buildPhase = ''
-      runHook preBuild
+    buildPhase =
+      if buildOffline
```

Comment:
`lib.optionalString` can be used instead of `if ... then ''..'' else ''..''` here.

Reply:
@wegank If `buildOffline` is true, I want to execute `mvn dependency:go-offline` only (without executing `mvn package`) and if `buildOffline` is false then I want to use the same logic as now (only executing `mvn package`) so that all current packages still work.

I currently fail to see how I can achieve this logic with `lib.optionalString`. Maybe you can help?

Reply:
@wegank Thank you for showing me how to achieve this with `lib.optionalString`.